packages/block-library/src/gallery/index.js

```diff
@@ -100,13 +100,28 @@ export const settings = {
 				blocks: [ 'core/image' ],
 				transform: ( attributes ) => {
 					const validImages = filter( attributes, ( { id, url } ) => id && url );
+					// init the align attribute from the first item which may be either the place holder or an image.
```

Comment:
I think we may be able to simplify the transform code to the following:
```
				transform: ( attributes ) => {
					// Init the align attribute from the first item which may be either the place holder or an image.
					let { align } = attributes[ 0 ];
					// Loop through all the images and check if they have the same align.
					align = every( attributes, [ 'align', align ] ) ? align : undefined;

					const validImages = filter( attributes, ( { id, url } ) => id && url );

					return createBlock( 'core/gallery', {
						images: validImages.map( ( { id, url, alt, caption } ) => ( { id, url, alt, caption } ) ),
						ids: validImages.map( ( { id } ) => id ),
						align,
					} );
				},
```.

With the added advantage that the loop to verify if the aligns are the same stops as soon as the first difference is found, while the reduce continue the iteration.

Reply:
That's a nice improvement, thanks! I've learned something today!